File: Training/Classic Puzzle - Easy/Temperature.cs
/**
 * @file Temperature.cs
 * @author Ladjouzi Rachid (RashidLadj@gmail.com)
 * @brief 
 * @version 0.1
 * @date 2021-10-11
 * 
 * @copyright Copyright (c) 2021
 */

using System;
using System.Linq;

class Solution{
    static void Main(){
        var n = Console.ReadLine();
        var min = 5526;

        if (n != "0"){
            var datas = Console.ReadLine().Split(' ').ToHashSet().ToArray();
            foreach (var data in datas){
                var d = int.Parse(data);
                if(Math.Abs(d) < Math.Abs(min) || (Math.Abs(d) == Math.Abs(min) && d > min))
                    min = d;
            }
        }

        Console.Write(n=="0"?0:min);
    }
}

File: Training/Classic Puzzle - Easy/Temperature.c
/**
 * @file Temperature.c
 * @author Ladjouzi Rachid (RashidLadj@gmail.com)
 * @brief 
 * @version 0.1
 * @date 2021-10-11
 * 
 * @copyright Copyright (c) 2021
 */

#define SMALLEST(x,y)((x)==0?(x):(y))

int main(){
    int n, min=5526; scanf("%d", &n);

    for (int i=0; i<n; ++i) {
        int t;  scanf("%d", &t);
        if (abs(min) > abs(t) || (abs(min) == abs(t) && t > min))
            min = t;
    }

    printf("%d",SMALLEST(n,min));
}